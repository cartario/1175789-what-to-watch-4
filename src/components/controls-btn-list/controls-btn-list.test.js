import React from 'react';
import {Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {history} from "../../history.js";

import ControlsBtnList from './controls-btn-list.jsx';

const currentMovie = {
  id: 1,
};

const mockStore = configureStore([]);

test(`should render control-btn-list`, () => {
  const store = mockStore({

  });
  const tree = renderer
    .create(
        <Router history = {history}>
          <Provider store = {store}>
            <ControlsBtnList
              addListClick = {() => {}}
              removeListClick = {() => {}}
              activeFilmId = {``}
              currentMovie = {currentMovie}
            />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
