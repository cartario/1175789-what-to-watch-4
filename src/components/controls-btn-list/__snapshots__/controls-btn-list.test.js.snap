// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render control-btn-list 1`] = `
Array [
  <a
    className="btn btn--play movie-card__button"
    href="/player/"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>,
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>,
]
`;
